src/sentry/dynamic_sampling/tasks/boost_low_volume_projects.py

```diff
@@ -101,6 +102,33 @@ def boost_low_volume_projects(context: TaskContext) -> None:
             boost_low_volume_projects_of_org.delay(org_id, projects_with_tx_count_and_rates)
 
 
+@instrumented_task(
+    name="sentry.dynamic_sampling.boost_low_volume_projects_of_org_with_query",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=25 * 60,
+    time_limit=2 * 60 + 5,
+    silo_mode=SiloMode.REGION,
+)
+@dynamic_sampling_task
+def boost_low_volume_projects_of_org_with_query(
+    org_id: OrganizationId,
+) -> None:
+    logger.info(
+        "boost_low_volume_projects_of_org_with_query",
+        extra={"traceparent": sentry_sdk.get_traceparent(), "baggage": sentry_sdk.get_baggage()},
+    )
+
+    task_name = _compute_task_name(boost_low_volume_projects_of_org_with_query.__name__)
+    context = TaskContext(task_name, MAX_TASK_SECONDS)
```

Comment:
Can we use decorator instead of calculating this manually?
`@dynamic_sampling_task_with_context(max_task_execution=MAX_TASK_SECONDS)`

This will pass the context as the keyword arg to the function

Reply:
Sure, but that does not yet allow passing additional parameters. We could improve `dynamic_sampling_task_with_context` to allow for that.

Reply:
Enhanced `dynamic_sampling_task_with_context` and using it with [bf2de6e](https://github.com/getsentry/sentry/pull/79582/commits/bf2de6e642914072b3967cf78ebb0c09a5b49ab8)